models/Venue.php

```diff
@@ -0,0 +1,16 @@
+<?php namespace October\Test\Models;
+use Model;
+
+class Venue extends Model
+{
+    public $table = 'october_test_venues';
+    public $timestamps = false;
+    public $attachOne = [
```

Comment:
I think what you need here is `morphOne`. This would resolve the issues. See Test number 5 in this plugin for an example.

Replace Venue with Plugin/Theme
Replace Fragment with Meta
Should be same result